geth/mailservice/mailservice.go

```diff
@@ -54,3 +51,7 @@ func (s *MailService) Start(server *p2p.Server) error {
 func (s *MailService) Stop() error {
 	return nil
 }
+
+func (s *MailService) nodeID() *ecdsa.PrivateKey {
+	return s.node.Server().PrivateKey
```

Comment:
it is also possible to use server that is sent to Start and remove node dependency